Sources/TuistAutomation/Utilities/TargetRunner.swift

```diff
@@ -0,0 +1,162 @@
+        }
+
+        switch (target.target.platform, target.target.product) {
+        case (.macOS, .commandLineTool):
+            try runExecutable(runnablePath, arguments: arguments)
+        case let (platform, .app):
+            try runApp(
+                target: target,
+                schemeName: schemeName,
+                configuration: configuration,
+                appPath: runnablePath,
+                platform: platform,
+                minVersion: minVersion,
+                version: version,
+                deviceName: deviceName,
+                arguments: arguments
+            )
+        default:
+            throw TargetRunnerError.runningNotSupported(target: target.target)
+        }
+    }
+
+    public func assertCanRunTarget(_ target: Target) throws {
+        switch (target.platform, target.product) {
+        case (.macOS, .commandLineTool),
+             (_, .app):
+            break
+        default:
+            throw TargetRunnerError.runningNotSupported(target: target)
+        }
+    }
+
+    private func runExecutable(_ executablePath: AbsolutePath, arguments: [String]) throws {
+        logger.notice("Running executable \(executablePath.basename)", metadata: .section)
+        logger.debug("Forwarding arguments: \(arguments.joined(separator: ", "))")
+        try System.shared.runAndPrint([executablePath.pathString] + arguments)
+    }
+
+    private func runApp(
+        target: ValueGraphTarget,
+        schemeName: String,
+        configuration: String,
+        appPath: AbsolutePath,
+        platform: Platform,
+        minVersion: Version?,
+        version: Version?,
+        deviceName: String?,
+        arguments: [String]
+    ) throws {
+        let buildTarget: XcodeBuildTarget = {
+            if target.project.xcodeProjPath.extension == ".xcworkspace" {
+                return XcodeBuildTarget.workspace(target.project.xcodeProjPath)
+            } else {
+                return XcodeBuildTarget.project(target.project.xcodeProjPath)
+            }
+        }()
+        let settings = try xcodeBuildController.showBuildSettings(buildTarget, scheme: schemeName, configuration: configuration)
+            .toBlocking()
+            .single()
+        let bundleId = settings[target.target.name]?.productBundleIdentifier ?? target.target.bundleId
+        let simulator = try simulatorController
+            .findAvailableDevice(platform: platform, version: version, minVersion: minVersion, deviceName: deviceName)
+            .toBlocking()
+            .single()
+
+        logger.debug("Running app \(appPath.pathString) with arguments [\(arguments.joined(separator: ", "))]")
+        logger.notice("Running app \(bundleId) on \(simulator.device.name)", metadata: .section)
+        try simulatorController.installApp(at: appPath, udid: simulator.device.udid)
+        try simulatorController.launchApp(bundleId: bundleId, udid: simulator.device.udid, arguments: arguments)
```

Comment:
One suggestion that I'd make here to improve the developer experience is turning the simulator controller API to be Reactive, for example using Combine, and do all the process of booting the simulator at the same time the app is being compiled. The time would look like:

-----------(xcodebuild)--------------
-----(launch simulator --------------|----- (install the app) -----(launch the app) --->

Reply:
It's a good opportunity to remove RxSwift from `SimulatorController` and use Combine instead.

Reply:
I like this, going to add this onto this PR unless others oppose.

Reply:
i think we can do it in a follow-up PR. Let's focus on getting this one working, and we add that as a future improvement.

Reply:
Sounds good, we can get the command out with this release